frontend/javascript/basket.js

```diff
@@ -0,0 +1,201 @@
+import { buildStockAvailabilityUrl } from './api-utils.js';
+import { createTemplate } from './products.js';
+
+let total = 0;
+const supportsTemplate = (function () {
+    return 'content' in document.createElement('template');
+})(); // IIFE
+
+function removeCartItem(templateClone2, currentProductId) {
+    let cartRemove_btn = templateClone2[templateClone2.length - 1].querySelector('.cart-remove');
+
+    cartRemove_btn.addEventListener('click', function () {
+        updateTotal('delete', currentProductId);
+        sessionStorage.removeItem(currentProductId);
+        cartRemove_btn.parentElement.remove();
+    });
+}
+
+function updateTotal(variation, currentProductId) {
+    let totalPrice = document.querySelector('.total__price');
+    let currentProductJSON = JSON.parse(sessionStorage.getItem(currentProductId));
+    let currentProductUnitCost = Number(currentProductJSON.unitPrice);
+    let currentProductTotalCost = Number(currentProductJSON.totalCost);
+    if (variation == 'increment') {
```

Comment:
Just adding a note for discussion (doesn't need to be updated if we're short on time): could this be a usecase for a [switch statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch)?